src/Analyzer.Utilities/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Semantics;
+
+namespace Analyzer.Utilities
+{
+    public static class IMethodSymbolExtensions
+    {
+        /// <summary>
+        /// Checks if the given method overrides Object.Equals.
+        /// </summary>
+        public static bool IsEqualsOverride(this IMethodSymbol method)
+        {
+            return method.IsOverride &&
+                   method.Name == WellKnownMemberNames.ObjectEquals &&
+                   method.ReturnType.SpecialType == SpecialType.System_Boolean &&
+                   method.Parameters.Length == 1 &&
+                   method.Parameters[0].Type.SpecialType == SpecialType.System_Object &&
+                   IsObjectMethodOverride(method);
+        }
+
+        /// <summary>
+        /// Checks if the given method overrides Object.GetHashCode.
+        /// </summary>
+        public static bool IsGetHashCodeOverride(this IMethodSymbol method)
+        {
+            return method.IsOverride &&
+                   method.Name == WellKnownMemberNames.ObjectGetHashCode &&  
+                   method.ReturnType.SpecialType == SpecialType.System_Int32 &&
+                   method.Parameters.Length == 0 &&
+                   IsObjectMethodOverride(method);
+        }
+
+        /// <summary>
+        /// Checks if the given method overrides Object.ToString.
+        /// </summary>
+        public static bool IsToStringOverride(this IMethodSymbol method)
+        {
+            return (method != null &&
+                    method.ReturnType.SpecialType == SpecialType.System_String &&
+                    method.Name == WellKnownMemberNames.ObjectToString &&
+                    method.Parameters.Length == 0 &&
+                    IsObjectMethodOverride(method));
+        }
+
+        /// <summary>
+        /// Checks if the given method overrides a method from System.Object
+        /// </summary>
+        private static bool IsObjectMethodOverride(IMethodSymbol method)
+        {
```

Comment:
So if I derived `A` from `object`, and I overrode `Equals`, and then I derived `B` from `A`, and I overrode `Equals` again, would this return `false`? Should it?